central/sensor/service/connection/connection_impl.go

```diff
@@ -206,14 +209,21 @@ func (c *sensorConnection) runSend(server central.SensorService_CommunicateServe
 			c.stopSig.SignalWithError(errors.Wrap(server.Context().Err(), "context error"))
 			return
 		case msg := <-c.sendC:
-			if err := server.Send(msg); err != nil {
+			if err := c.sendWithSizingMetric(server, msg); err != nil {
```

Comment:
Nit, I think this is cleaner to have a wrapper like the streams and just implement the server interface

Reply:
Done